### ISSUE TITLE
Compile issues with 1.10 on MacOS High Sierra

### ISSUE THREAD

**Issue by tom111** [Macaulay2/M2#646]
Hi,
I'm starting this new issue to debug my compile issues with 1.10. After the discussion in a previous issue I'm now using "Apple LLVM version 8.1.0 (clang-802.0.42" as the compiler and GNU make version 4.2.1, configuring as follows

```
alias make=gmake
 ../../configure --enable-download
make
```

With this the compile runs through, but the resulting binary is defective as shown by various errors while running examples / creating documentation. There seem to be race conditions involved, as I can produce different errors by running make again after the first error.  Here is one:

```
--making example results for frac in file ../../../usr-dist/common/share/doc/Macaulay2/Macaulay2Doc/example-output/_frac.out
ulimit -t 700; ulimit -m 850000; ulimit -v 850000; ulimit -s 8192; cd /var/folders/73/jfd7r2r15zs03ks2lnx6zfpm0000gn/T/M2-41703-0/2-rundir/; /Users/tom/temp/M2/M2/BUILD/tom/usr-dist/x86_64-Darwin-MacOS-10.12.5/bin/M2-binary --silent --print-width 77 --stop --int --no-readline -q --no-randomize <"/var/folders/73/jfd7r2r15zs03ks2lnx6zfpm0000gn/T/M2-41703-0/1_frac.m2" >>"/Users/tom/temp/M2/M2/BUILD/tom/usr-dist/common/share/doc/Macaulay2/Macaulay2Doc/example-output/_frac.errors" 2>&1
sh: line 1: 41735 Abort trap: 6           /Users/tom/temp/M2/M2/BUILD/tom/usr-dist/x86_64-Darwin-MacOS-10.12.5/bin/M2-binary --silent --print-width 77 --stop --int --no-readline -q --no-randomize < "/var/folders/73/jfd7r2r15zs03ks2lnx6zfpm0000gn/T/M2-41703-0/1_frac.m2" >> "/Users/tom/temp/M2/M2/BUILD/tom/usr-dist/common/share/doc/Macaulay2/Macaulay2Doc/example-output/_frac.errors" 2>&1
../../../usr-dist/common/share/doc/Macaulay2/Macaulay2Doc/example-output/_frac.errors:0:1: (output file) error: Macaulay2 exited with status code 134
/var/folders/73/jfd7r2r15zs03ks2lnx6zfpm0000gn/T/M2-41703-0/1_frac.m2:0:1: (input file)
M2: *** Error 134
currentString:1:1:(3):[3]: error: 1 error(s) occurred running examples for package Macaulay2Doc
gmake[2]: *** [Makefile:91: /Users/tom/temp/M2/M2/BUILD/tom/usr-dist/x86_64-Darwin-MacOS-10.12.5/lib/Macaulay2/x86_64-Darwin-MacOS-10.12.5/Macaulay2Doc/.installed] Error 1
```

I've seen another where the M2 file for the Projective Hilbert polynomial fails, but I can't reproduce it now. I vaguely remember seeing failures in that file with earlier versions of M2 that were compiled with the wrong mpir, or was it mpir compiled with clang, or something. I don't remember, sorry.

**Comment by tom111** [comment]
Let me append another issue here that I ran into on a new Mac with HighSierra. If readline is installed by the M2 system, the patches are not in order:

=== making .patched-6.3 in readline
if [ -d build -a -d build-old/build ] ; \
    then echo please remove or move build-old/build out of the way ; \
         exit 1 ; \
    fi
if [ -d build ] ; then mv build build-old ; fi
mkdir build
cd build && gtar xf /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline-6.3.tar.gz --gzip
cd build/readline-6.3 && for i in /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-001 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-002 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-003 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-004 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-005 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-006 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-007 /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-008 ; do echo patching with $i ; patch -p0 < $i ; done
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-001
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-002
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-003
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-004
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-005
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-006
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-007
patch: **** Only garbage was found in the patch input.
patching with /Users/tom/M2/M2/BUILD/tom/../../BUILD/tarfiles/readline63-008
patch: **** Only garbage was found in the patch input.
gmake[2]: *** [../Makefile.library:167: .patched-6.3] Error 2
gmake[2]: Leaving directory '/Users/tom/M2/M2/BUILD/tom/libraries/readline'
gmake[1]: *** [Makefile:7: all-in-readline] Error 2
gmake[1]: Leaving directory '/Users/tom/M2/M2/BUILD/tom/libraries'
gmake: *** [GNUmakefile:191: all-in-libraries] Error 2
gmake: Leaving directory '/Users/tom/M2/M2/BUILD/tom'

**Comment by tom111** [comment]
Ah, let's close this and make a new issue, sorry.
